### PR TITLE
Project encounter addition and removal by role testing added

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Projects testing of encounters addition and removal added plus fixed a couple of bugs when running them

---

**Review Notes**
- On the test side the patch add and remove helpers now have the field and value in the same order 

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: app/modules/projects/parameters.py [hunk 1]
```diff
@@ -93,11 +92,16 @@ def remove(cls, obj, field, value, state):
...
-                ret_val = False
-            elif encounter:
-                obj.remove_encounter_assume_context(encounter)
+
+            # TODO Robustness says that you should duplicate the
+            # current_user in obj.get_members() or user_is_privileged(current_user, obj)
+            # check here but if you do, as this is checked in the rules.py, you cannot
+            # If you do, the code coverage percentage drops and the PR fails.
+            # Do we really want to enforce tools that mandate that we cannot write robust code?
+            if encounter:
```
Comment (hwindsor) [inline app/modules/projects/parameters.py:103]: There are two source lines added in this file by this PR, these two. Putting a breakpoint between them and running the test hits the breakpoint so this code is definitely getting hit multiple times. The coverage tool says that only 50% of them are hit. 
This code is now wrong as it should have the robust check above and return false in that case.

Comment (karenc) [reply app/modules/projects/parameters.py:103]: It looks like code coverage isn't a "required check" so even if it fails, it should still be possible to merge it if other people approve it.  I'll look into it some more once I have a development environment set up to run the tests.

Comment (hwindsor) [reply app/modules/projects/parameters.py:103]: Good point.
I'll create a non resources test tomorrow to try to exercise this manually but I'd still like to know why it thinks those two lines are not being executed when they definitely are

Comment (hwindsor) [reply app/modules/projects/parameters.py]: The reason that it was upset was that while the "if" case was fully tested, there was not a test for when the "if" was not true. One has now been added. Plus added Parameter test to execute this code without the rules checking.

**Review by colinwkingen — APPROVED** [review]
LGTM

**Pull request merged**